nx/lib/nx/lin_alg.ex

```diff
@@ -967,4 +967,53 @@ defmodule Nx.LinAlg do
       opts
     )
   end
+
+  @doc """
+  Produces the tensor taken to the given power by dot-product.
+
+  The input is always a square tensor and a non-negative integer,
+  and the output is a square tensor of the same dimensions as the input tensor.
+
+  # Examples
+      iex> Nx.LinAlg.matrix_power(Nx.tensor([[1, 2], [3, 4]]), 0)
+      #Nx.Tensor<
+        s64[2][2]
+        [
+          [1, 0],
+          [0, 1]
+        ]
+      >
+
+      iex> Nx.LinAlg.matrix_power(Nx.tensor([[1, 2], [3, 4]]), 6)
+      #Nx.Tensor<
+        s64[2][2]
+        [
+          [5743, 8370],
+          [12555, 18298]
+        ]
+      >
+
+      iex> Nx.LinAlg.matrix_power(Nx.eye(3), floor(:math.pow(2, 32))-1)
```

Comment:
I think this might be a bit much to include in a doctest. I would instead put this in `nx_test`.

Reply:
It's not very slow though. It just takes 64-ish dot-products of an identity matrix. I'll go time it for you.

Reply:
On my box it takes 511µs.

Reply:
Sorry, I did not mean too much in terms of performance. What I meant was as an example in the docs it's a little noisy when compared to the other 2 examples. While it is a simple test, on the surface it may seem confusing or complex for someone who is just trying to understand what this function does. Does that make sense?

Reply:
Ah, yes. I can put `65535` into it or so. That way the noise is removed, while much of the spirit of the test remains?